### PR TITLE
User Interection System

### PR DESCRIPTION
**Pull request by AndrePu**
1> Post-methods Sub/Unsub/Ban/Unban added
2> sub/unsub buttons on profile page added

### CODE REVIEW

**Review by AleksMorozova — APPROVED** [review]

File: Cooper/ClientApp/src.v2/app/components/user-info/user-info.component.html [hunk 1]
```diff
@@ -14,4 +14,8 @@ <h5>E-mail:</h5>
     <h5>Bio:</h5>
     <p class="card-field">{{profile.description}}</p>
   </mat-card-content>
+
+  <div (click)="(isSubscription) ? onUnsubscribeButtonClicked() : onSubscribeButtonClicked()">
+    <button mat-raised-button class="my-button">{{(isSubscription)? 'Unsubscribe' : 'Subscribe'}}</button>
+  </div>
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/components/user-info/user-info.component.html:24]: it is better to make two separate buttons to make it more clear and easier to maintain

File: Cooper/ClientApp/src.v2/app/layouts/profile/profile.layout.ts [hunk 2]
```diff
@@ -95,6 +96,8 @@ export class ProfileLayoutComponent implements OnInit, OnDestroy {
     }
 
     this.isOwnProfile = this.sessionService.GetSessionUserId() === this.profile.id;
+
+    this.isSubscription = this.isOwnProfile ? true : this.sessionService.isSubscription(this.profile.id);
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/layouts/profile/profile.layout.ts]: this.isSubscription = this.isOwnProfile || this.sessionService.isSubscription(this.profile.id);

File: Cooper/ClientApp/src.v2/app/layouts/profile/profile.layout.ts [hunk 3]
```diff
@@ -118,4 +121,69 @@ export class ProfileLayoutComponent implements OnInit, OnDestroy {
       this.userService.updateUserInfo(updatedUser);
     }
   }
+
+  public subscribe(userId: number): void {
+    this.usersSocialConnectionsService.subscribe(userId).subscribe(
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/layouts/profile/profile.layout.ts]: why two subscribe and not clear what it is
too much functions with name subscribe

File: Cooper/ClientApp/src.v2/app/layouts/profile/profile.layout.html [hunk 4]
```diff
@@ -18,6 +18,9 @@
           <coop-user-info
           *ngIf = "!isOwnProfile"
           [profile] = "profile"
+          [isSubscription] = "isSubscription"
+          (subscribe)="subscribe($event)"
+          (unsubscribe)="unsubscribe($event)"
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/layouts/profile/profile.layout.html]: event handler better to name with prefix "on"

File: Cooper/ClientApp/src.v2/app/layouts/profile/profile.layout.ts [hunk 5]
```diff
@@ -118,4 +121,69 @@ export class ProfileLayoutComponent implements OnInit, OnDestroy {
...
+            }
+          }
+
+          if (index !== this.friends.length) {
+            this.friends.splice(index, 1);
+            this.friendsAmount--;
+          }
+        }
+      });
+  }
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/layouts/profile/profile.layout.ts:187]: i am not sure that this code should be in component

File: Cooper/ClientApp/src.v2/app/services/session/session.service.ts [hunk 6]
```diff
@@ -32,6 +36,28 @@ export class SessionService {
...
+    const user = new User();
+    user.id = this.sessionProfile.id;
+    user.email = this.sessionProfile.email;
+    user.description = this.sessionProfile.description;
+    user.name = this.sessionProfile.name;
+    user.nickname = this.sessionProfile.nickname;
+    user.photoURL = this.sessionProfile.photoURL;
+
+    return user;
+  }
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/session/session.service.ts:49]: it looks strange. it should be other model, or we need converter, or we need constructor or type casting

**Review by horba — APPROVED** [review]
it seems it has architecture problems. we can discus in future and make refactoring
Please, make simple changes

**Pull request merged**